vllm/v1/sample/ops/topk_topp_sampler.py

```diff
@@ -224,8 +227,8 @@ def apply_top_k_only(
     max_top_k = k.max()
     # topk.values tensor has shape [batch_size, max_top_k].
     # Convert top k to 0-based index in range [0, max_top_k).
-    k_index = k.sub_(1).unsqueeze(1).expand(logits.shape[0], 1)
-    top_k_mask = logits.topk(max_top_k, dim=1).values.gather(1, k_index.long())
+    k_index = k.sub_(1).unsqueeze(1)
+    top_k_mask = logits.topk(max_top_k, dim=1).values.gather(1, k_index)
```

Comment:
Is this unintentional? This seems to be reverting what's at HEAD.

Reply:
yep this change was applied on top of @njhill original implementation when we first intended to use that function for TPU too.
Now that implementations have diverged, I am reverting to what we had before.

Reply:
I see. We might want to keep `k_index.long()` as that's not TPU-related. It was added in #15907.

Reply:
Right good catch, thanks for checking!